components/core/src/clp/DictionaryReader.hpp

```diff
@@ -152,25 +131,27 @@ void DictionaryReader<DictionaryIdType, EntryType>::close() {
         throw OperationFailed(ErrorCode_NotInit, __FILENAME__, __LINE__);
     }
 
-    m_segment_index_decompressor.close();
-    m_segment_index_file_reader.close();
-    m_dictionary_decompressor.close();
-    m_dictionary_file_reader.close();
-
     m_num_segments_read_from_index = 0;
     m_entries.clear();
 
     m_is_open = false;
 }
 
 template <typename DictionaryIdType, typename EntryType>
-void DictionaryReader<DictionaryIdType, EntryType>::read_new_entries() {
-    if (false == m_is_open) {
-        throw OperationFailed(ErrorCode_NotInit, __FILENAME__, __LINE__);
-    }
+void DictionaryReader<DictionaryIdType, EntryType>::read_new_entries(
+        std::string const& dictionary_path,
+        std::string const& segment_index_path
+) {
+    constexpr size_t cDecompressorFileReadBufferCapacity = 64 * 1024;  // 64 KB
+
+    FileReader dictionary_file_reader{dictionary_path};
```

Comment:
put them here instead of right in front of the code that uses them, so that if either file fails to open, we can skip the processing.